File: org.kevoree.modeling.microframework/src/main/java/org/kevoree/modeling/api/extrapolation/ExtrapolationStrategies.java
package org.kevoree.modeling.api.extrapolation;

/**
 * Created by duke on 10/28/14.
 */
public enum ExtrapolationStrategies {

    DISCRETE(new DiscreteExtrapolationStrategy()), LINEAR_REGRESSION(new LinearRegressionExtrapolationStrategy()), POLYNOMIAL(new PolynomialExtrapolationStrategy());

    private ExtrapolationStrategies(ExtrapolationStrategy wrappedStrategy) {
        this.extrapolationStrategy = wrappedStrategy;
    }

    private ExtrapolationStrategy extrapolationStrategy;

    public ExtrapolationStrategy strategy() {
        return extrapolationStrategy;
    }

}

File: org.kevoree.modeling.microframework.typescript/src/main/ts/org/kevoree/modeling/api/strategy/ExtrapolationStrategies.ts

class ExtrapolationStrategies {

  public static DISCRETE: ExtrapolationStrategies = new ExtrapolationStrategies(new DiscreteExtrapolationStrategy());
  public static LINEAR_REGRESSION: ExtrapolationStrategies = new ExtrapolationStrategies(new LinearRegressionExtrapolationStrategy());
  public static POLYNOMIAL: ExtrapolationStrategies = new ExtrapolationStrategies(new PolynomialExtrapolationStrategy());
  private extrapolationStrategy: ExtrapolationStrategy = null;

  constructor(wrappedStrategy: ExtrapolationStrategy) {
    this.extrapolationStrategy = wrappedStrategy;
  }

  public strategy(): ExtrapolationStrategy {
    return this.extrapolationStrategy;
  }

  public equals(other: AccessMode): boolean {
        return this == other;
    }

}

